### PR TITLE
servlet: github API

### PR DESCRIPTION
**Pull request by evacchi**
This servlet requires a bit more care than #10, so I'll do it in phases. It is a partial port from https://github.com/modelcontextprotocol/servers/tree/main/src/github but I am rewriting it in Go.

Currently supported: 

Issues: 

- `gh-create-issue` Create issue
- `gh-get-issue` Get issue
- `gh-update-issue` Add comments to an issue
- `gh-add-issue-comment` Read the contents of an issue

Files:

- `gh-get-file-contents` Get contents and metadata about a file on a branch
- `gh-create-or-update-file` Create or a update a file on a branch
- `gh-push-files` Bulk push files to a branch

Branches: 

- `gh-create-branch` Create a new branch
- `gh-create-pull-request` Create a PR from a branch

### CODE REVIEW

**Comment by evacchi** [comment]
this is mostly working, but there is still some issue I need to figure out with getting file contents from a branch

**Comment by evacchi** [comment]
I think the set of features is nice enough now, I'll do more updates to clean up the code and add more features later, in the meantime, this is working and allows some nice integrations

File: servlets/github/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+Files:
+
+- `gh-get-file-contents` Get contents and metadata about a file on a branch
+- `gh-create-or-update-file` Create or a update a file on a branch
+- `gh-push-files` Bulk push files to a branch
+
+Branches: 
+
+- `gh-create-branch` Create a new branch
+- `gh-create-pull-request` Createa a PR from a branch
```
Comment (evacchi) [inline servlets/github/README.md:20]:
```suggestion
- `gh-create-pull-request` Create a PR from a branch
```

File: servlets/github/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# GitHub Servlet
+
+Currently supported: 
+
+Issues: 
+
+- `gh-get-issue` Create issues
```
Comment (nilslice) [inline servlets/github/README.md:7]:
```suggestion
- `gh-get-issue` Get issue
```

**Review by nilslice — APPROVED** [review]
Just a readme change, otherwise LGTM!

**Pull request merged**